clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -1225,6 +1231,14 @@ raw2trace_t::write(void *tls, const trace_entry_t *start, const trace_entry_t *e
                                 reinterpret_cast<const char *>(end) -
                                     reinterpret_cast<const char *>(start)))
         return "Failed to write to output file";
+    // If we're at the end of a block (minus its delayed branch) we need
+    // to split now to avoid going too far by waiting for the next instr.
+    if (tdata->cur_chunk_instr_count >= chunk_instr_count_) {
```

Comment:
Do we need this if statement if we are asserting that they are `==`? Or is `DEBUG_ASSERT` lets you proceed even when they are not equal?

Reply:
DEBUG_ASSERT is a nop in release build.  The > part is being defensive.